### ISSUE TITLE
Unable to use reportImportCycles

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#4388]
reportImportCycles doesn't work as I would have expected (with latest version of pyright 1.1.286). With a sample project such as this:
```
models/
---> __init__.py
---> model_a.py
---> model_b.py
---> pyproject.toml
```

_model_a.py_
```python

from __future__ import annotations

from .model_b import B


class A:
    a = B()
```

_model_b.py_
```python
from __future__ import annotations

from .model_a import A


class B:
    b = A()
```

_pyproject.toml_
```python
[tool.pyright]
include = ["./**"]
reportImportCycles=true
```

if `__init__.py ` file is empty, pyright correctly reports `models/model_a.py: error: Cycle detected in import chain`, but if `__init__.py` contains the following (which is quite common in python projects)

```python
from __future__ import annotations

from .model_a import A
from .model_b import B

__all__ = ["A", "B"]
```

Then pyright doesn't report any error, while the circular dependency is unchanged.

I this a bug ? Or I'm i missing smth ?

**Comment by erictraut** [comment]
Thanks for the great bug report. This was a bug I introduced about a year ago when adding an optimization to the cycle detection logic. I'm surprised no one has reported it since then.

It will be fixed in the next release.

**Comment by mehdigmira** [comment]
Awesome ! Looking forward to testing this in a new version 😄

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.287, which I just published. It will also be included in a future release of pylance.
